app/models/manageiq/providers/nuage/network_manager/event_catcher/stream.rb

```diff
@@ -0,0 +1,47 @@
+require 'qpid_proton'
+
+class ManageIQ::Providers::Nuage::NetworkManager::EventCatcher::Stream
+  include Vmdb::Logging
+
+  def self.test_amqp_connection(options = {})
+    # Ensure we just test the connection. AMQP channel will be established and
+    # started, however it will be immediately stopped.
+    options[:test_connection] = true
+    begin
+      container = connect(options)
+      container.run
+      true
+    rescue => e
+      _log.info("#{log_prefix} Failed connecting to ActiveMQ: #{e.message}")
+      raise
+    end
+  end
+
+  def self.log_prefix
+    "MIQ(#{self.class.name})"
+  end
+
+  def initialize(options = {})
+    @options = options
+  end
+
+  def start(&message_handler_block)
+    _log.debug("#{self.class.log_prefix} Opening amqp connection using options #{@options}")
+    @options[:message_handler_block] = message_handler_block if message_handler_block
+    connection.run
+  end
+
+  def stop
+    @handler.stop
```

Comment:
I understand `@handler.stop` sets `@stop_listening = true` which is then read in `on_message` function of the MessagingHandler. I'm wondering if this is sufficient - if no new message arrives after stopping, then the @handler will keep running.

I'm not familiar with qpid proton gem, but would it work if we use something like:

```ruby
@handler.interrupt
```

to stop the event catcher? See [this](https://github.com/gberginc/qpid_proton_gem/blob/6188f0732382fc7983f5c0aa8f6da379050d71fd/lib/messenger/messenger.rb#L399-L413) documentation. Please note, it's just an idea of mine, not sure if it's doable.

Reply:
Problem with this is that it's again part of the `messenger`, not the `reactor`. The reactor was chosen for this use case as it seems more lightweight as we are only listening to the events.

I understand your concern and will investigate the reactor to see if there is similar way to stop  the handler

Reply:
@miha-plesko Check [this commit](https://github.com/ManageIQ/manageiq-providers-nuage/pull/9/commits/f5707bc13f6a3142390abbe0236c3d6f1b097cc4) please. This now automatically closes the connection when stop is requested. Thanks for pointing out the problem.

Reply:
I like it better now, thanks for this change.